src/api/documentsApi.ts

```diff
@@ -0,0 +1,70 @@
+import { spWebContext } from "api/SPWebContext";
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+
+/**
+ * Gets all requests
+ */
+export const useDocuments = (requestId: number) => {
+  return useQuery({
+    queryKey: ["documents", requestId],
+    queryFn: () => getDocuments(requestId),
+  });
+};
+
+const getDocuments = async (requestId: number) => {
+  const path = "requests/" + requestId;
+  return spWebContext.web
+    .getFolderByServerRelativePath(path)
+    .files.select(
+      "Name",
+      "TimeLastModified",
+      "ServerRelativeUrl",
+      "ModifiedBy",
+      "ModifiedBy/Id",
+      "ModifiedBy/EMail",
+      "ModifiedBy/Title",
+      "UniqueId"
+    )
+    .expand("ModifiedBy")
+    .orderBy("Name")<SPDocument[]>();
+};
+
+export const useDeleteDocument = (document: SPDocument) => {
+  const queryClient = useQueryClient();
+  return useMutation(
+    ["deleteDocument", document.Name],
+    async () => {
+      return spWebContext.web.getFileById(document.UniqueId).recycle();
+    },
+    {
+      onSuccess: async () => {
+        queryClient.invalidateQueries(["documents"]);
```

Comment:
If we had the requestId, we could invalidate just this requests documents, instead of all the cache.  Probably not a big deal, but an option.

Reply:
I was thinking along the same lines, so sounds good to me.